src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.EasyRequest.cs

```diff
@@ -221,17 +221,6 @@ public void FailRequest(Exception error)
                 // and doing it here prevents reuse by an intermediate handler sitting between the client
                 // and this handler.
 
-                // However, if we got an original position for the request stream, we seek back to that position,
-                // for the corner case where the stream does get reused before it's disposed by the HttpClient
-                // (if the same request object is used multiple times from an intermediate handler, we'll be using
-                // ReadAsStreamAsync, which on the same request object will return the same stream object, which
-                // we've already advanced).
```

Comment:
Can we be confident that this isn't an issue that we're going to run into? As mentioned in the bug report WinHttpHandler does not clear the location of the stream, but does it have some other method for dealing with reused streams? Or is this just not a concern in practice?

Reply:
I modified the test to use the same content stream, all handlers behave the same, they don't reset the position of the stream to the beginning, they send 0 content second time.

Reply:
>I modified the test to use the same content stream, all handlers behave the same, they don't reset the position of the stream to the beginning, they send 0 content second time.

This might be true for an actual second SendAsync() call using the same HttpContent (StreamContent) object.  However, during a single SendAsync() call, it is sometimes necessary to re-send the request body, i.e. for POSTs when an initial 401 happens.  In those cases, the handler (such as WinHttpHandler) will "rewind" the StreamContent to send it again.